opentracing-shim/test/shim_utils_test.cc

```diff
@@ -0,0 +1,280 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "shim_mocks.h"
+
+#include "opentelemetry/opentracingshim/shim_utils.h"
+
+#include "opentracing/tracer.h"
+
+#include <gtest/gtest.h>
+
+namespace trace_api = opentelemetry::trace;
+namespace baggage   = opentelemetry::baggage;
+namespace common    = opentelemetry::common;
+namespace nostd     = opentelemetry::nostd;
+namespace shim      = opentelemetry::opentracingshim;
+
+TEST(ShimUtilsTest, IsBaggageEmpty)
+{
+  auto none = nostd::shared_ptr<baggage::Baggage>(nullptr);
+  ASSERT_TRUE(shim::utils::isBaggageEmpty(none));
+
+  auto empty = nostd::shared_ptr<baggage::Baggage>(new baggage::Baggage({}));
+  ASSERT_TRUE(shim::utils::isBaggageEmpty(empty));
+
+  std::map<std::string, std::string> list{{"foo", "bar"}};
+  auto non_empty = nostd::shared_ptr<baggage::Baggage>(new baggage::Baggage(list));
+  ASSERT_FALSE(shim::utils::isBaggageEmpty(non_empty));
+}
+
+TEST(ShimUtilsTest, StringFromValue)
+{
+  ASSERT_EQ(shim::utils::stringFromValue(true), "true");
+  ASSERT_EQ(shim::utils::stringFromValue(false), "false");
+  ASSERT_EQ(shim::utils::stringFromValue(1234.567890), "1234.567890");
+  ASSERT_EQ(shim::utils::stringFromValue(42l), "42");
+  ASSERT_EQ(shim::utils::stringFromValue(55ul), "55");
+  ASSERT_EQ(shim::utils::stringFromValue(std::string{"a string"}), "a string");
+  ASSERT_EQ(shim::utils::stringFromValue(opentracing::string_view{"a string view"}),
+            "a string view");
+  ASSERT_EQ(shim::utils::stringFromValue(nullptr), "");
+  ASSERT_EQ(shim::utils::stringFromValue("a char ptr"), "a char ptr");
+
+  opentracing::util::recursive_wrapper<opentracing::Values> values{};
+  ASSERT_EQ(shim::utils::stringFromValue(values.get()), "");
+
+  opentracing::util::recursive_wrapper<opentracing::Dictionary> dict{};
+  ASSERT_EQ(shim::utils::stringFromValue(dict.get()), "");
+}
+
+TEST(ShimUtilsTest, AttributeFromValue)
+{
+  auto value = shim::utils::attributeFromValue(true);
+  ASSERT_EQ(value.index(), common::AttributeType::kTypeBool);
+  ASSERT_TRUE(nostd::get<bool>(value));
+
+  value = shim::utils::attributeFromValue(false);
+  ASSERT_EQ(value.index(), common::AttributeType::kTypeBool);
+  ASSERT_FALSE(nostd::get<bool>(value));
+
+  value = shim::utils::attributeFromValue(1234.567890);
+  ASSERT_EQ(value.index(), common::AttributeType::kTypeDouble);
+  ASSERT_EQ(nostd::get<double>(value), 1234.567890);
+
+  value = shim::utils::attributeFromValue(42l);
+  ASSERT_EQ(value.index(), common::AttributeType::kTypeInt64);
+  ASSERT_EQ(nostd::get<int64_t>(value), 42l);
+
+  value = shim::utils::attributeFromValue(55ul);
+  ASSERT_EQ(value.index(), common::AttributeType::kTypeUInt64);
+  ASSERT_EQ(nostd::get<uint64_t>(value), 55ul);
+
+  value = shim::utils::attributeFromValue(std::string{"a string"});
+  ASSERT_EQ(value.index(), common::AttributeType::kTypeString);
+  ASSERT_EQ(nostd::get<nostd::string_view>(value), nostd::string_view{});
```

Comment:
`std::string{"a string"}` should equal `nostd::string_view{"a string"}` rather than `nostd::string_view{}`?

Reply:
Agree, an overlook on my side. I had temporarily removed this mapping since string_view doesn't own the underlying data from the passed in string and would be flagged as undefined behavior by the address sanitizer check if passed by const reference.